azavea/franklin
# Add Tile Info Endpoints

## Overview

This adds support for providing Tile Info endpoints for STAC items that have a COG present. The idea is that we can dynamically generate tiles for any COG asset with GeoTrellis Server.

### Checklist


### Notes

 - Need to open a new issue for smoke tests/integration tests to better test the API configuration at runtime
 - The COG media type is slightly incorrect based on documentation in the STAC repository, I deferred fixing that at the moment because it requires updating `stac4s`
 - We need to add the `tiles` relation type to `stac4s` as well
 - In the process of trying to import someone else's catalog I realized that one assumption for the import is that it has at least _one_ collection and every item references a collection. I think we should start enforcing some of these restrictions. Without enforcing them we happily import data, but it is impossible to navigate to any items.

### Testing Instructions

- Import a catalog from S3: `bloop run application -- import --catalog-root s3://rasterfoundry-development-data-us-east-1/berlin-catalog/catalog.json --db-host localhost`
- Start server with tiles: `./scripts/server --with-tiles`
- Go to `http://localhost:9090/collections/berlin/items/image` -- verify that there is a tiles link
- Click on the tiles link and compare it to the output described in [this](https://docs.google.com/document/d/1r6LP2amldq75w93KpX7O17ix1SBoe6lzBTNzKWSRlyc/edit) document
 - Stop server and start it without the tiles URL: `./scripts/server` - verify there aren't tile links and the direct link to the tile info 404s

Closes #137
